cmd/nubuild/cli_test.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/util/cmdrunner"
+	"github.com/nuclio/nuclio/test/suite"
+)
+
+type CliSuite struct {
+	suite.NuclioTestSuite
+
+	imageName string
+}
+
+func (suite *CliSuite) buildCli() {
+	options := &cmdrunner.RunOptions{WorkingDir: &suite.NuclioRootPath}
+	_, err := suite.Cmd.Run(options, "go build ./cmd/nubuild")
+	suite.Require().NoError(err, "Can't build nubuild")
+}
+
+func (suite *CliSuite) generateImageName() string {
+	buf := make([]byte, 4)
+	rand.Seed(time.Now().Unix())
+	_, err := rand.Read(buf)
+	suite.Require().NoError(err, "Create read random bytes")
+	return fmt.Sprintf("nubuild-test-%x", buf)
+}
+
+func (suite *CliSuite) SetupSuite() {
```

Comment:
The tests currently have a function order convention:
1. SetupSuite
2. SetupTest <-- btw, please check if you don't mean SetupTest in this module, it's supposed to be much more common than SetupSuite as it re-initializes state between tests
3. TearDownTest
4. TearDownSuite
5. Helper functions
6. Tests

Then at the very end the testing thingie that calls testify. I'll add an issue to write code a conventions document.